docs/Software Engineering Blog/posts/Comparing_Human_and_AI_way_for_Big-String-Set_Intersection_Performance_in_Java_by_JMH.md

```diff
@@ -0,0 +1,115 @@
+
+### how to use
+
+There are lots of guidance or How-To articles online to do that. I follow the README in [jmh-gradle-plugin](https://github.com/melix/jmh-gradle-plugin) and every thing go well. If you are interested in the detail or code implementation, please refer to [my repository](https://github.com/NoahHsu/performance-benchmark/tree/main/src/jmh/java/org/example/bigset/intersacation), there is a working example.
+
+### scenario description
+
+In my use case, I need to extract user IDs that are present in both of two sets: one representing users who have previously purchased my product, and the other representing users who have agreed to receive advertisements.
+
+To evaluate performance, I designed two scenarios:
+
+1. Both sets contain 100,000 UUIDs, with 1,000 IDs as the intersection.
+2. One set contains 100,000 UUIDs, while the other contains 1,100 UUIDs, including the same 1,000 IDs as the intersection. 
+
+## Comparing Result
+
+Here, we can start to execute the JMH command by the [jmh-gradle-plugin](https://github.com/melix/jmh-gradle-plugin). However, there are some essential points that we should be cautious about.
+
+### Be Careful the Lazy Loading Value
+
+Some of the method in Java follow the lazy loading way (e.g. guava's `Sets.intersection()` only return a `SetView` instance until calling some method of that SetView). These kinds of method will be executed extremely quick. while other methods need to take between 0.045 ms ~ 0.085 ms, the lazy loading way cost only around 0.00001 ms.
+
+![lazyloading.png](resources/BigSetIntersec/lazyloading.png)
+
+So we need to add some common method call after it. For example, adding a value assignment from the intersection set helps to trigger the SetView to truly execute the intersection, and we finally get the correct result.
+
+```java
+    public static <T> Set<T> getIntersection(Set<T> set1, Set<T> set2) {
+        Set<T> answer;
+        if (set1.size() <= set2.size()) {
+            answer =  Sets.intersection(set1, set2);
+        } else {
+            answer = Sets.intersection(set2, set1);
+        }
+
+        int size = answer.size();
+
+        return answer;
+    }
+```
+
+### JVM Warm-Up Matters
+
+Next, we take a look on how JVM warm up affect the stability of the result. If only one warm-up once, we will get high errors, which means the result is not stable.
+![1Warnup.png](resources/BigSetIntersec/1Warmup.png)
+
+For comparison, if we warm up for three iteration, we can get a more stabilize result with lower error.
+
+![3Warmup.png](resources/BigSetIntersec/3Warmup.png)
+
+## Summary
+
+Check on the final result, the AI generated method in working, but may not be the best solution. We still got some chances to optimize and modify the given answer. Especially for the coding detail like the creation of new instances actually make the time cost twice. So if we can ensure the parameters will be `HashSet` or do it only when they aren't. The performance can be improved. 
```

Comment:
📝 **[vale]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>[write-good.E-Prime] Try to avoid using 'be'.